gst/datarepo/gstdatareposink.c

```diff
@@ -502,24 +497,42 @@ gst_data_repo_sink_get_caps (GstBaseSink * bsink, GstCaps * filter)
   GstCaps *caps = NULL;
 
   GST_OBJECT_LOCK (sink);
-  caps = gst_pad_get_pad_template_caps (GST_BASE_SINK_PAD (bsink));
-  GST_OBJECT_UNLOCK (sink);
+  caps = sink->fixed_caps;
 
   GST_INFO_OBJECT (sink, "Got caps %" GST_PTR_FORMAT, caps);
-
-  if (caps && filter) {
-    GstCaps *intersection =
-        gst_caps_intersect_full (filter, caps, GST_CAPS_INTERSECT_FIRST);
-
-    gst_caps_unref (caps);
-    caps = intersection;
+  if (caps) {
+    if (filter)
+      caps = gst_caps_intersect_full (filter, caps, GST_CAPS_INTERSECT_FIRST);
+    else
+      caps = gst_caps_ref (caps);
   }
 
   GST_DEBUG_OBJECT (sink, "result get caps: %" GST_PTR_FORMAT, caps);
+  GST_OBJECT_UNLOCK (sink);
 
   return caps;
 }
 
+/**
+ * @brief Set whether the given pad caps are static or not.
+ */
+static void
+gst_data_repo_sink_set_is_static_tensors (GstDataRepoSink * sink)
+{
+  GstStructure *structure;
+  GstTensorsConfig config;
+
+  g_return_if_fail (sink != NULL);
+  g_return_if_fail (sink->fixed_caps != NULL);
+  g_return_if_fail (sink->data_type == GST_DATA_REPO_DATA_TENSOR);
+
+  structure = gst_caps_get_structure (sink->fixed_caps, 0);
+  gst_tensors_config_from_structure (&config, structure);
+  sink->is_static_tensors = gst_tensors_config_is_static (&config);
+  gst_tensors_config_free (&config);
+  GST_ERROR ("is_static_tensors = %d", sink->is_static_tensors);
```

Comment:
log level : gst-info or debug?

Reply:
Wow, I will remove it, thank you. :smile: